test/integration/bind_integration_test.cc

```diff
@@ -0,0 +1,121 @@
+#include <string>
+
+#include "common/http/header_map_impl.h"
+#include "common/protobuf/utility.h"
+
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class BindIntegrationTest : public BaseIntegrationTest,
+                            public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  BindIntegrationTest() : BaseIntegrationTest(GetParam()) {}
+
+  void SetUp() override {
+    // Fake upstream.
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP1, version_));
+
+    // Admin access log path and address.
```

Comment:
FWIW I find the following substantially harder to read vs. YAML/JSON. Is there any particular reason to build the config this way? (I know we have talked about making integration tests better, but to me that's mostly about reducing code duplication in tests. For configs IMO it's a lot simpler to read a config as is, even if there is duplication).

Reply:
I don't think it would be bad to have a "default config" in string JSON/YAML as long as we're (eventually) reading it into in-memory structs and allowing C++ style manipulation before re-serializing.

Reply:
+1, there is no reason why tests couldn't manipulate configs once read in. IMO that would be much easier to read/grok.

Reply:
I was trying to demonstrate how we can dynamically build from scratch a complete config. I think we will want to do this to reduce boilerplate in integration tests long term. I'll add a TODO on potentially factoring part of this out to a static file.

The idea is we will only have this kind of construction happening in one `IntegrationTestConfig` class that is appropriately parameterized, we won't be copying+pasting this across tests.